test/integration/scheduler/framework_test.go

```diff
@@ -989,12 +1032,39 @@ func TestPrebindPlugin(t *testing.T) {
 			fail:   true,
 			reject: true,
 		},
+		{
+			name:           "fail on 1st try but succeed on retry",
+			fail:           true,
+			reject:         false,
+			succeedOnRetry: true,
```

Comment:
here we are testing that the incoming Pod doesn't go to the unschedulable queue and gets retried, right?

Reply:
It actually goes to unschedulableQ for the 1st try. And then the preBind logic fails itself, so it's able to be moved back to activeQ. And after that, as `succeedOnRetry` is true, it's scheduled successfully on 2nd try.

Reply:
> It actually goes to unschedulableQ for the 1st try

Why? Doesn't it directly fail on PreBind?

Reply:
Nope with succeedOnRetry = true:

```go
func (pp *PreBindPlugin) PreBind(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) *framework.Status {
	pp.numPreBindCalled++
	if _, tried := pp.podUIDs[pod.UID]; tried && pp.succeedOnRetry {
		return nil
	}
	pp.podUIDs[pod.UID] = struct{}{}
```